### ISSUE TITLE
Mac Apple tcl/tk crashes program when caret ^ character typed in IDLE or tkinter

### ISSUE THREAD

**Issue by 2956bcb0-1e2f-4653-8fc5-ababe5896f84** [python/cpython#75619]
BPO | [31438](https://bugs.python.org/issue31438)
--- | :---
Nosy | @terryjreedy, @ronaldoussoren, @ned-deily, @bitdancer, @kkoyun
Superseder | <li>bpo-24170: IDLE crashes when I press ^  caret key</li>
Files | <li>[IDLE 3 crash error log.rtf](https://bugs.python.org/file47137/IDLE%203%20crash%20error%20log.rtf "Uploaded as text/rtf at 2017-09-13.00:08:56 by @kkoyun"): Example error log generated by IDLE 3 crash</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = <Date 2017-09-14.18:22:47.108>
created_at = <Date 2017-09-13.00:08:57.452>
labels = ['OS-mac', 'type-crash']
title = 'Mac Apple tcl/tk crashes program when caret ^ character typed in IDLE or tkinter'
updated_at = <Date 2017-09-16.16:55:45.431>
user = 'https://github.com/kkoyun'
```

bugs.python.org fields:
```python
activity = <Date 2017-09-16.16:55:45.431>
actor = 'terry.reedy'
assignee = 'none'
closed = True
closed_date = <Date 2017-09-14.18:22:47.108>
closer = 'karakoyun'
components = ['macOS']
creation = <Date 2017-09-13.00:08:57.452>
creator = 'karakoyun'
dependencies = []
files = ['47137']
hgrepos = []
issue_num = 31438
keywords = []
message_count = 11.0
messages = ['302012', '302026', '302031', '302080', '302135', '302136', '302194', '302250', '302342', '302344', '302345']
nosy_count = 5.0
nosy_names = ['terry.reedy', 'ronaldoussoren', 'ned.deily', 'r.david.murray', 'karakoyun']
pr_nums = []
priority = 'normal'
resolution = 'duplicate'
stage = 'resolved'
status = 'closed'
superseder = '24170'
type = 'crash'
url = 'https://bugs.python.org/issue31438'
versions = ['Python 3.6']
```

</p></details>

**Comment by 2956bcb0-1e2f-4653-8fc5-ababe5896f84** [comment]
IDLE 3 crashes and quits whenever I type the caret (^) character in the editor. I have to copy and paste the caret character into the IDLE editor whenever I have to use the caret character. I have the IDLE 3.6.2 running on macOS Sierra (v10.12.6) on mid-2011 27" iMac. Attached is an example  error log generated when IDLE 3 crashes after the typing of the caret character.

**Comment by bitdancer** [comment]
Have you followed the instructions on this page?:

https://www.python.org/download/mac/tcltk/

**Comment by terryjreedy** [comment]
In the future, please upload plain text.  I am fairly certain that this is a duplicate of previous issues about Apple's buggy tcl/tk causing crashes  that bypass Python's error reporting.

**Comment by 2956bcb0-1e2f-4653-8fc5-ababe5896f84** [comment]
I downloaded and installed TCL 8.6.6.8606 and restarted my computer. I verified that the new version was installed. Unfortunately the problem has not been solved. Thank you.

Alper

**Comment by terryjreedy** [comment]
Ned, is 3.6.2 tkinter on Mac supposed to work with tcl/tk 8.6.6?

**Comment by terryjreedy** [comment]
If your installed Python with the 64/32 Mac installer from python.org, follow the instruction, on the link given, to install Active 8.5.18 and \*not* 8.6.  I believe the issue is that the \_tkinter binary is compiled to work with 8.5, not 8.6.

If you used any other installer, talk with its providers.

**Comment by 2956bcb0-1e2f-4653-8fc5-ababe5896f84** [comment]
Problem has been solved after installation of ActiveTcl 8.5.18.0 from ActiveState's web site as per Terry’s recommendations and, then, reinstallation of Python 3.6.2 package from python.org. However, there were many (mis)steps per my ignorance and laziness in between. Thank you very much for all the help!

Noteworthy case notes:

1. The previously installed IDLE 3.6.2 as part of the Python 3.6.2 package continued crashing whenever I entered the caret character after the installation of ActiveTcl 8.5.18.0. I noticed that the IDLE’s Tk version showed 8.5.9 even after I installed, as recommended, ActiveTcl 8.5.18.0, and rebooted my computer. Note that the “info patchlevel” script I executed within tclsh had started returning “8.5.18” instead of “8.5.9” after the installation of ActiveTcl 8.5.18.0.

2. I could not find a way to make the existing Python / IDLE 3.6.2 installation recognise ActiveTcl 8.5.18.0 as the IDLE’s Tk version kept showing 8.5.9. Therefore, I  reinstalled the Python 3.6.2 package downloaded from python.org. The package made the installation to a new location rather than overwriting the existing Python / IDLE 3.6.2 files.

3. The newly installed IDLE 3.6.2 has been working fine without showing any signs of instability. There are no crashes when I type the caret character. Its Tk version shows 8.5.18 along with the tclsh interpreter. 

4. I removed all the files associated with the “buggy” Python / IDLE 3.6.2 installation via the terminal window in order to prevent any future confusion between the old and new installations.

5. Though I installed ActiveTcl 8.6.6.8606 from ActiveState’s web site in my earlier trials, I don’t think I was able to fully test whether this latest Tcl really works with Python / IDLE 3.6.2 or not. After installing ActiveTcl 8.6.6.8606, the “info patchlevel” I executed within tchlsh started returning “8.6.6” but the (old) IDLE 3.6.2’s Tk version still showed 8.5.9. I, then, installed ActiveTcl 8.5.18.0 after uninstalling ActiveTcl 8.6.6.8606 as a precaution. It  occurred to me that I may need to reinstall the Python 3.6.2 package from python.org only after installing the ActiveTcl 8.5.18.0.

**Comment by terryjreedy** [comment]
Glad it is working and thanks for the detail notes.

Ned, is there anything we should change on the web page?  Perhaps emphasize more not to upgrade to tcl/tk 8.6?  Is it usual to have to re-install python?

Alper, please do not revert headers after a core dev has changed them.  This was never about IDLE, except insofar as IDLE is a victim of the problem.  IDLE is a tkinter program and you could have written a tkinter program yourself with a similar problem.

**Comment by 2956bcb0-1e2f-4653-8fc5-ababe5896f84** [comment]
I believe it would be better if IDLE was somehow mentioned in the header. This is partly the reason I failed to recognize that this was a problem encountered before.

new title suggestion: Apple tcl/tk crashes and causes IDLE to quit when caret character typed in IDLE

**Comment by terryjreedy** [comment]
OK.  This is more complete, still succinct enough, and still true.

**Comment by terryjreedy** [comment]
The 'all text' is apparently limited to 2 words.  'idle crash caret' in the title box gets this issue and the superseder above.  Move to all-text box and nothing is returned.  Delete 1 of 3 words and get multiple answers.
